components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/codegenerator/CodeGenerator.java

```diff
@@ -0,0 +1,726 @@
+                .append(generateTables(siddhiApp.getTableList()))
+                .append(generateWindows(siddhiApp.getWindowList()))
+                .append(generateTriggers(siddhiApp.getTriggerList(), siddhiApp.getSourceList(),
+                        siddhiApp.getSinkList()))
+                .append(generateAggregations(siddhiApp.getAggregationList()))
+                .append(generateFunctions(siddhiApp.getFunctionList()))
+                .append(generateQueries(siddhiApp.getQueryLists()))
+                .append(generatePartitions(null));
+
+        return siddhiAppStringBuilder.toString();
+    }
+
+    /**
+     * Generates a string representation of the siddhi app name and description annotations
+     *
+     * @param appName        The app name
+     * @param appDescription The app description
+     * @return The string representation of the siddhi app name and description annotations
+     */
+    private String generateAppNameAndDescription(String appName, String appDescription) {
+        StringBuilder appNameAndDescriptionStringBuilder = new StringBuilder();
+
+        if (appName != null && !appName.isEmpty()) {
+            appNameAndDescriptionStringBuilder.append(SiddhiStringBuilderConstants.APP_NAME_ANNOTATION)
+                    .append(appName)
+                    .append(SiddhiStringBuilderConstants.SINGLE_QUOTE)
+                    .append(SiddhiStringBuilderConstants.CLOSE_BRACKET)
+                    .append(SiddhiStringBuilderConstants.NEW_LINE);
+        } else {
+            appNameAndDescriptionStringBuilder.append(SiddhiStringBuilderConstants.DEFAULT_APP_NAME_ANNOTATION)
+                    .append(SiddhiStringBuilderConstants.NEW_LINE);
+        }
+
+        if (appDescription != null && !appDescription.isEmpty()) {
+            appNameAndDescriptionStringBuilder.append(SiddhiStringBuilderConstants.APP_DESCRIPTION_ANNOTATION)
+                    .append(appDescription)
+                    .append(SiddhiStringBuilderConstants.SINGLE_QUOTE)
+                    .append(SiddhiStringBuilderConstants.CLOSE_BRACKET)
+                    .append(SiddhiStringBuilderConstants.NEW_LINE);
+        } else {
+            appNameAndDescriptionStringBuilder.append(SiddhiStringBuilderConstants.DEFAULT_APP_DESCRIPTION_ANNOTATION)
+                    .append(SiddhiStringBuilderConstants.NEW_LINE);
+        }
+
+        appNameAndDescriptionStringBuilder.append(SiddhiStringBuilderConstants.NEW_LINE);
+
+        return appNameAndDescriptionStringBuilder.toString();
+    }
+
+    /**
+     * Generates a string representation of all the streams in a Siddhi app
+     *
+     * @param streamList A list of StreamConfig objects from the SiddhiAppConfig object
+     * @param sourceList A list of sources from the SiddhiAppConfig object
+     * @param sinkList   A list of sinks from the SiddhiAppConfig object
+     * @return The string representation of the stream definitions
```

Comment:
@moizalicious Please update the java doc with throws exception in similar places.